jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormData.java

```diff
@@ -295,9 +402,47 @@ public Parser(String boundary, MultiPartCompliance multiPartCompliance, Complian
             parser = new MultiPart.Parser(Objects.requireNonNull(boundary), compliance, listener);
         }
 
+        public void parse(Content.Source content, Promise<Parts> immediate, Promise.Invocable<Parts> future)
```

Comment:
Not 2 promises for the same result!

It is just an implementation detail that we do not want to surface to user APIs.